iterators.tex

```diff
@@ -342,55 +342,59 @@
 \tcode{shared_ptr<int>} are \tcode{Readable} and have an associated value type. But a smart pointer
 like \tcode{shared_ptr<void>} is not \tcode{Readable} and has no associated value type.\exitnote
 
-\rSec2[iterators.movewritable]{Concept MoveWritable}
+\rSec2[iterators.writable]{Concept Writable}
 
 \pnum
-The \tcode{MoveWritable} concept describes the requirements for moving a value into an iterator's
+The \tcode{\oldtxt{Move}Writable} concept describes the requirements for \oldtxt{moving}\newtxt{writing} a value into an iterator's
 referenced object.
 
-\indexlibrary{\idxcode{MoveWritable}}%
+\indexlibrary{\idxcode{Writable}}%
 \begin{codeblock}
   template <class Out, class T>
-  concept bool MoveWritable() {
+  concept bool @\oldtxt{Move}@Writable() {
     return Semiregular<Out>() &&
-      requires (Out o, T v) {
-        *o = std::move(v); // not required to be equality preserving
+      requires(Out o, T@\newtxt{\&\&}@ v) {
+        *o = std::@\oldtxt{move}\newtxt{forward<T>}@(v); // not required to be equality preserving
       };
   }
 \end{codeblock}
 
 \pnum
-Let \tcode{v} be an rvalue or a lvalue of type (possibly \tcode{const}) \tcode{T}, and let \tcode{o}
-be a dereferenceable object of type \tcode{Out}. Then \tcode{MoveWritable<Out, T>()} is satisfied if and only if
+Let \tcode{v} be an \oldtxt{rvalue or a lvalue of type (possibly \tcode{const})}\newtxt{an expression such that \tcode{decltype(v)} is} \tcode{T}, and let \tcode{o}
+be a dereferenceable object of type \tcode{Out}. Then \tcode{\oldtxt{Move}Writable<Out, T>()} is satisfied if and only if
 
 \begin{itemize}
-\item If \tcode{Out} is \tcode{Readable},
-then after the assignment \tcode{*o = std::move(v)}, \tcode{*o} is equal
+\item If \oldtxt{\tcode{Out} is }\tcode{Readable<Out>()\newtxt{ \&\& Same<value_type_t<Out>, decay_t<T>{>}()}}\newtxt{ is satisfied},
+then\newtxt{ \tcode{*o}} after the assignment\oldtxt{ \tcode{*o = std::move(v)}, \tcode{*o}} is equal
```

Comment:
What I would *like* to require is that there must exist a function `f(decay_t<T>)` such that `*o = v` inserts an element into the output sequence with value `f(v)`, but it seems too ridiculously abstract to bother.